public/characterCreator.html

```diff
@@ -1 +1,157 @@
+        <input type="number" id="lvl" name="lvl" value="1" class=""><br><br>
+        <label>Stat Points</label>
+        <input type="number" id="statpoints" name="statpoints" value="0" class=""><br><br>
+        <label>Experience</label>
+        <input type="number" id="xp" name="xp" value="0" class="xp"><br><br>
+        <label>IQ</label>
+        <input type="number" id="iq" name="iq" value="0" class="stats mental"><br><br>
+        <label>Mental Endurance</label>
+        <input type="number" id="mEndurance" name="mEndurance" value="0" class="stats mental"><br><br>
+        <label>Mental Affinity</label>
+        <input type="number" id="mAffinity" name="mAffinity" value="0" class="stats mental"><br><br>
+        <label>Physical Strength</label>
+        <input type="number" id="pStrength" name="pStrength" value="0" class="stats physical"><br><br>
+        <label>Physical Prowess</label>
+        <input type="number" id="pProwess" name="pProwess" value="0" class="stats physical"><br><br>
+        <label>Physical Endurance</label>
+        <input type="number" id="pEndurance" name="pEndurance" value="0" class="stats physical"><br><br>
+        <label>Speed</label>
+        <input type="number" id="speed" name="speed" value="0" class="stats speed"><br><br>
+        <label>Wealth</label>
+        <input type="number" id="wealth" name="wealth" value="2" class="wealth"><br><br>
+        <label>Power Name</label>
+        <input type="text" id="powername" name="powername" class="power"><br><br>
+        <label>Power Description</label>
+        <input type="text" id="powerdescription" name="powerdescription" class="power"><br><br>
+        <input type="submit" value="Create">
+    </form>
+    <button onclick="preview()">Preview</button>
+    <script type="text/javascript">
+        const form = document.getElementById("characterCreator");
+        form.addEventListener("input", function() {
+            preview();
+        });
+
+        function preview() {
+            const elements = document.getElementsByClassName("stats");
+            document.getElementById('deficitpoints').innerHTML = calculateDeficit(elements)
+        }
+        /** Returns the cost of statpoints for the set level of the stat. Returns a negative number to indicate a gain of statpoints
+         * @param {number} setstat The level the stat is set to
+         * @return {number} result This is the resulting cost of statpoints for the set stat, negative if its a gain of statpoints
+         */
+        function generateStatCost(setStatLevel) {
+            setStatLevel = parseInt(setStatLevel);
+            if (setStatLevel < 0) { //Checks if setstatlevel is negative if so returns the negative setstatlevel as a statpoint gain
+                return setStatLevel
+            } else {
+                for (loopIterator = 0, result = 0; loopIterator <= setStatLevel; loopIterator++) { // iterates to find the statpoint cost needed to get a stat to the setstat level
```

Comment:
maybe look into this, you don't need to use it
https://scotch.io/tutorials/understanding-memoization-in-javascript#toc-a-functional-approach